From 1c19bf8cc294e95c8de314cc457bcea6854c3a2d Mon Sep 17 00:00:00 2001
From: Alexey Neyman <stilor@att.net>
Date: Tue, 28 Feb 2017 09:29:21 -0800
Subject: [PATCH 2/2] Macos does not have <elf.h> and needs a local copy

Also, move <elf/xtensa.h> up - generic <elf.h> does not have
definitions for xtensa relocations.

Local file, cygwin-elf.h, needs to include <stdint.h> for standard
integer types which is POSIX - rather than glibc-originated <features.h>.

Signed-off-by: Alexey Neyman <stilor@att.net>
---
 cygwin-elf.h | 2 +-
 elf2flt.c    | 7 +++++--
 2 files changed, 6 insertions(+), 3 deletions(-)

diff --git a/cygwin-elf.h b/cygwin-elf.h
index bd72b37..8e3dbff 100644
--- a/cygwin-elf.h
+++ b/cygwin-elf.h
@@ -26,7 +26,7 @@ typedef uint16_t u_int16_t;
 typedef uint32_t u_int32_t;
 typedef uint64_t u_int64_t;
 #else
-#include <features.h>
+#include <stdint.h>
 #endif
 /* Standard ELF types.  */
 
diff --git a/elf2flt.c b/elf2flt.c
index 08296cf..19a4d4d 100644
--- a/elf2flt.c
+++ b/elf2flt.c
@@ -58,14 +58,17 @@ const char *elf2flt_progname;
 #include <elf/h8.h>      /* TARGET_* ELF support for the BFD library            */
 #elif defined(TARGET_arm)
 #include <elf/arm.h>
-#elif defined(__CYGWIN__) || defined(__MINGW32__) || defined(TARGET_nios) || defined(TARGET_nios2)
-#include "cygwin-elf.h"	/* Cygwin uses a local copy */
 #elif defined(TARGET_xtensa)
 #include <elf/xtensa.h>	/* TARGET_* ELF support for the BFD library */
+#elif defined(TARGET_nios) || defined(TARGET_nios2)
+#include "cygwin-elf.h" // <elf/nios2.h> does not have R_NIOS_* declarations
 #elif defined(TARGET_microblaze)
 #include <elf/microblaze.h>	/* TARGET_* ELF support for the BFD library */
 #elif defined(TARGET_v850)
 #include <elf/v850.h>
+#elif (__CYGWIN__) || defined(__MINGW32__) || defined(__APPLE__)
+// FIXME: does Cygwin need this? It has <elf.h> in /usr/include
+#include "cygwin-elf.h" // Some systems don't have <elf.h>
 #else
 #include <elf.h>      /* TARGET_* ELF support for the BFD library            */
 #endif
-- 
2.9.3

